# Research

## 2024 – Current



### <span style="font-size: 26px; font-weight: 500;">Influenza Vaccination Impact on DNA Methylation in UGA6 Cohort</span>

**Graduate Student Researcher** | Supervised by [Dr. Matteo Pellegrini](https://medschool.ucla.edu/people/matteo-pellegrini-phd) in his [Lab](https://www.pellegrini.mcdb.ucla.edu/)

- Influenza vaccination, a key preventive measure against annually recurring pandemics, elicits immune responses potentially regulated by DNA methylation. Prior studies within University of Georgia cohorts (UGA3, 4, and 5) have highlighted the role of DNA methylation in modulating the [RIG‐I signaling pathway](https://www.nature.com/articles/s41598-024-51665-9) and [T cell innate antiviral pathways](https://clinicalepigeneticsjournal.biomedcentral.com/articles/10.1186/s13148-024-01730-x). This study validates these findings in the larger UGA6 cohort (n = 254) to identify epigenetic factors influencing immune responses to influenza vaccination. Multivariate linear regression models with leave-one-out cross-validation (LOOCV) were used to assess associations between DNA methylation profiles and sero-protection levels at day 0 and day 28, as well as sero-conversion at day 28. Significant CpG sites were mapped to genes and pathways through gene set enrichment analysis, identifying the RIG‐I signaling pathway as the key regulator of vaccine-induced immunity. Parallel predictive models for RNA-seq and cytokine datasets further elucidated molecular mechanisms. Additionally, genome-wide expression quantitative trait loci (eQTL) analyses linked DNA methylation to gene expression, and correlations between DNA methylation and cytokine data highlighted additional regulatory insights. Our study provides a comprehensive understanding of the epigenetic regulation of immune responses to influenza vaccines and identifies potential therapeutic targets in a large cohort setting, contributing to advancements in precision medicine for influenza vaccination. 

![](img/UGA6.png)


### <span style="font-size: 26px; font-weight: 500;">Nonlinear Aging Dynamics in Co-methylation Network</span>

**Graduate Student Researcher**  | Supervised by [Dr. Feng Gao](https://ph.ucla.edu/about/faculty-staff-directory/feng-gao)


- Non-linear epigenetic models have emerged as critical regulators of aging, with recent findings highlighting their predominant influence on nervous system genes and bivalently marked chromatin regions, in contrast to well-established linear models like [Horvath's aging clock](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-10-r115). This study investigates the role of [non-linear methylation](https://www.nature.com/articles/s41467-024-47316-2) dynamics in aging mice colon tissue and their associations with gene expression and pathways. Using [metilene](https://pmc.ncbi.nlm.nih.gov/articles/PMC4728377/), we identified differentially methylated regions (DMRs) across five age groups and constructed graph-based networks with [NetworkX](https://en.wikipedia.org/wiki/NetworkX), uncovering non-linear co-methylation clusters at chromosomal, individual, and age-group levels. Functional associations were explored with [Cytoscape](https://en.wikipedia.org/wiki/Cytoscape), mapping genes to biological processes, cellular components, pathways, and human orthologs. Gene set enrichment analysis revealed dynamic changes in gene functional groups during aging, consistent with non-linear methylation patterns. These findings enhance our understanding of the epigenetic mechanisms driving aging and highlight the significance of non-linear dynamics in age-related biological processes.

![](img/mice_colon_1.png)
![](img/mice_colon_2.png)



### <span style="font-size: 26px; font-weight: 500;">Inferring Immune Histories from SARS-CoV-2 RNA-Seq Data</span>

**Warren Alpert Fellow**  | Supervised by [Dr. Vanessa D Jonsson](https://jonssonlab.com/people)


- Understanding immune memory and response to SARS-CoV-2 infection is essential for advancing COVID-19 treatment, personalized treatment and vaccine strategies. This study employs SARS-CoV-2 RNA-seq data from nasal swabs to infer immune histories through an integrated analysis pipeline. Immune cell proportions are estimated using [CIBERSORT](https://www.nature.com/articles/nmeth.3337), while metagenomics annotation characterizes microbial protein landscapes. [TRUST4](https://www.nature.com/articles/s41592-021-01142-2) is used to infer TCR and BCR repertoires, with internal tools performing isotype analysis and clustering based on sequence and biochemistry (ESM/other, tcrdist3). BCR phylogenies are reconstructed using [Immcantation](https://immcantation.readthedocs.io/en/stable/), and viral epitope binding to HLA is predicted with netMHCpan, incorporating HLA typing via HLA arcasHLA and MHC Hammer. Finally, we compare the immune histories inferred from the RNA-seq data with publicly available [SARS-CoV2 antibodies and nanobodies data](https://opig.stats.ox.ac.uk/webapps/covabdab/) to validate the pipeline. 

![](img/Immune_history.png){width=70% fig-align="center"}



## 2021 – 2023

### <span style="font-size: 26px; font-weight: 500;">Statistical Data Analysis for Biology and Biomedicine Research</span>

Undergraduate Research Assistant 

- **Modeling Transcription Factor Diffusion with Stochastic Processes**:
  + Simulated transcription factor (TF) diffusion in the nucleus using Markov Chain Monte Carlo (MCMC) methods, modeling TF movement as a time-homogeneous Markov process. The model quantified transient states (non-specific DNA interactions) and recurrent states (specific binding at regulatory elements).
- **Multivariate Regression Analysis in Agricultural Research**:
  + Conducted multivariate regression analysis in R to evaluate the relationship between rice yield and irrigation at five critical growth stages (seedling, tillering, panicle initiation, flowering, and grain filling). Identified optimal irrigation strategies to maximize yield while minimizing resource consumption. Incorporated additional variables such as soil type, weather conditions, and nutrient input for a comprehensive analysis.

## 2020 – 2021

### <span style="font-size: 26px; font-weight: 500;">Review: Lifestyle Factors Affecting Semen Quality</span>

- This review examined the impact of dietary habits, BMI, alcohol, smoking, stress, mobile phone usage, and microplastics on sperm parameters. It highlighted that a balanced diet, exercise, and stress management improve semen quality, while smoking, alcohol, and microplastic exposure impair it. Statistical models like multivariate regression and meta-analysis were used to assess fertility impacts, emphasizing the public health importance of addressing these lifestyle factors.

![](img/semen_quality.png){width=50% fig-align="center"}


## 2019 – 2020

### <span style="font-size: 26px; font-weight: 500;">Lymphocyte Regeneration after Thymocyte Damage in Zebrafish [\[presentation slides\]](assets/LYMPHOCYTE_REGENERATION_AFTER_THYMOCYTE_DAMAGE_IN ZEBRAFISH.pdf)</span>

Volunteer | Supervised by [Dr. Li Li](https://people.ucas.ac.cn/~0064236?language=en)

- Thymus atrophy and T lymphocyte deficiencies are common in aging and following damage caused by factors such as radiation, chemotherapy, or bacterial infections, leading to impaired immune function. Our study investigates the regeneration of T lymphocytes in zebrafish after thymocyte injury, leveraging the transparency and genetic tractability of zebrafish as a model organism. A damage model was constructed by treating [Tg(coro1a:Dendra2-NTR)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9503817/) transgenic zebrafish with 2 mM Metronidazole for 12 hours, resulting in significant lymphocyte reduction and apoptosis in the thymus, as confirmed by whole-mount in situ hybridization ([WISH](https://pubmed.ncbi.nlm.nih.gov/25218374/)) and confocal microscopy.

- Through real-time PCR ([RT-qPCR](https://www.ncbi.nlm.nih.gov/probe/docs/techqpcr/)), we discovered that the [CXCL12/CXCR4 chemokine signaling pathway](https://www.ijbs.com/v19p3341.htm) plays a critical role in the regeneration of thymic lymphocytes. Inhibition of this pathway with [WZ811](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5040679/), a CXCL12/CXCR4 receptor antagonist, significantly impaired lymphocyte recovery, suggesting that CXCL12/CXCR4 signaling is essential for lymphocyte regeneration. Additionally, lineage tracing experiments revealed that most regenerating lymphocytes originated from hematopoietic stem cells migrating from the kidney to the thymus. These findings highlight the importance of the CXCL12/CXCR4 pathway in lymphocyte regeneration and offer insights into the mechanisms of immune recovery after thymocyte damage.

![](img/REGENERATION.png)



------------------------------------------------------------------------

# Publication

- **Ai, J.** (2021). Lifestyle does affect semen quality: factor determination in a statistical way. *Lecture Notes in Medicine and Life Sciences*, 4, 163--171. [Read Article](https://www.clausiuspress.com/conference/article/artId/7114.html)
- **Ai, J.**, Pickering, H., Pellegrini, M. et al. Influenza Vaccination: DNA Methylation and Immune Response Mechanisms. *Genome Biology*. (forthcoming).
- **Ai, J.**, Fan, Y., Gao, F. et al. Nonlinear aging dynamics in co-methylation network. *Cell Reports*. (forthcoming).





